cuqi/model/_model.py

```diff
@@ -35,14 +35,18 @@ class Model(object):
         The Jacobian of the forward operator with respect to the forward operator input,
         evaluated at a point (`wrt`). The signature of the Jacobian function should be (`wrt`).
         The Jacobian function should return a 2D ndarray of shape (range_dim, domain_dim).
-        The Jacobian function is used to specify the gradient function (vector-Jacobian product)
+        The Jacobian function is used to specify the gradient function (direction-Jacobian product)
```

Comment:
This is good point to link it with direction in our terminology. One suggestion is to keep vector (since the term vector-Jacobian product VJP seems to be known ) but mention that we refer to the vector as direction here, maybe something like
```suggestion
        The Jacobian function is used to specify the gradient function by computing the vector-Jacobian
        product (VJP), here we refer to the vector in the VJP as the `direction` since it is the direction at 
        which the gradient is computed.
```